### PR TITLE
feat(web): Zenter email signup configuration

### PR DESCRIPTION
**Pull request by RunarVestmann**
# Zenter email signup configuration

## What

* Added a way to subscribe to zenter mailing lists

## Why

* This was a feature needed by the directorate of fisheries

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by brynjarorng — APPROVED** [review]
LGTM(Devops)

File: libs/api/domains/email-signup/src/lib/serviceProvider.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Provider } from '@nestjs/common'
+import { ConfigType, LazyDuringDevScope } from '@island.is/nest/config'
+import { EmailSignupConfig } from './emailSignup.config'
+import { EmailSignupService } from './emailSignup.service'
+
+export const EmailSignupServiceProvider: Provider<EmailSignupService> = {
```
Comment (eirikurn) [inline libs/api/domains/email-signup/src/lib/serviceProvider.ts:6]: We use this "Provider" pattern to wrap classes that have been generated for us, like openapi clients.

In this case, you can mark EmailSignupService directly with `@Injectable()` and  `@Scope(LazyDuringDevScope)` (writing from memory).

File: libs/api/domains/email-signup/src/lib/emailSignup.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    if (emailSignupModel.signupType === 'zenter') {
+      return this.emailSignupService.subscribeToZenter(
+        emailSignupModel,
+        inputFields,
+      )
+    }
+
+    return { subscribed: false }
```
Comment (eirikurn) [inline libs/api/domains/email-signup/src/lib/emailSignup.resolver.ts:49]: Could move most/all of this to `this.emailSignupService.subscribe(input)`. Might even refactor mailchimp and zenter specific logic to their own services (eg ZenterSignupService) which EmailSignupService depends on.

**Review by eirikurn — APPROVED** [review]
Approving with minor suggestions.

File: libs/api/domains/email-signup/src/lib/emailSignup.resolver.ts (hunk 2 above)
Comment (RunarVestmann) [reply libs/api/domains/email-signup/src/lib/emailSignup.resolver.ts]: I did it 🙌

File: libs/api/domains/email-signup/src/lib/serviceProvider.ts (hunk 1 above)
Comment (RunarVestmann) [reply libs/api/domains/email-signup/src/lib/serviceProvider.ts:6]: I think I managed to do it in a recent commit I pushed just now

**Review by ingolfurs — APPROVED** [review]

**Pull request merged**